src/protocol/damage.rs

```diff
@@ -114,14 +114,17 @@ pub struct BadDamageError {
     pub sequence: u16,
 }
 impl TryParse for BadDamageError {
-    fn try_parse(remaining: &[u8]) -> Result<(Self, &[u8]), ParseError> {
+    fn try_parse(initial_value: &[u8]) -> Result<(Self, &[u8]), ParseError> {
+        let remaining = initial_value;
         let (response_type, remaining) = u8::try_parse(remaining)?;
         let (error_code, remaining) = u8::try_parse(remaining)?;
         let (sequence, remaining) = u16::try_parse(remaining)?;
         if response_type != 0 {
             return Err(ParseError::ParseError);
         }
         let result = BadDamageError { error_code, sequence };
+        let remaining = remaining.get(32usize.saturating_sub(initial_value.len() - remaining.len())..)
```

Comment:
Does this need `saturating_sub`? I'd instead propose `assert!(initial_value.len() >= remaining.len());` and then doing a plain subtraction. Or is this assert "stupid enough" to warrant being a `debug_assert`?

Reply:
Actually... `assert!(initial_value.len() - remaining.len() < 32)` and just do `let remaining = initial_value.get(32..).ok_or(ParseError::ParseError)?;`?